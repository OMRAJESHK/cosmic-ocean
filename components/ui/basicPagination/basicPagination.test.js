import React from "react";
import { screen, render, fireEvent } from "@testing-library/react";
import BasicPagination from "./basicPagination";

const mockOnBack = jest.fn();
const mockOnNext = jest.fn();

describe("Basic Pagination Testing", () => {
  it("should render Basic Pagination Component", () => {
    const component = (
      <BasicPagination onBack={mockOnBack} onNext={mockOnNext} />
    );
    render(component);
    expect(component).toMatchSnapshot();
  });

  it("should call a callback when Back and Next buttons is pressed.", () => {
    render(<BasicPagination onBack={mockOnBack} onNext={mockOnNext} />);
    const backButton = screen.getByText(/Back/, { exact: true });
    const nextButton = screen.getByText(/Next/, { exact: true });

    expect(backButton).toBeInTheDocument();
    fireEvent.click(backButton);
    expect(mockOnBack).toHaveBeenCalledTimes(1);

    expect(nextButton).toBeInTheDocument();
    fireEvent.click(nextButton);
    expect(mockOnNext).toHaveBeenCalledTimes(1);
  });
});
