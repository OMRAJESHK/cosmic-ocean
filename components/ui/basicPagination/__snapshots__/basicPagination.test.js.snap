// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Pagination Testing should render Basic Pagination Component 1`] = `
<BasicPagination
  onBack={[MockFunction]}
  onNext={[MockFunction]}
/>
`;
